src/tree/render.js

```diff
@@ -4,17 +4,24 @@ import box from './box';
 import createPaths from './path';
 import transform from './transform';
 
-export const filterTree = ({ topId, isExpanded, expandedChildren }, nodeTree) => {
+export const filterTree = ({ topId, isExpanded, expandedChildren }, nodeTree, extended) => {
   const topNode = nodeTree.descendants().find(node => node.data.id === topId) || nodeTree;
   const subTree = [];
   if (isExpanded && topNode.children) {
     // children
     topNode.children.forEach(child => {
       subTree.push(child);
-      if (child.children && expandedChildren.indexOf(child.data.id) !== -1) {
-        child.children.forEach(grandChild => {
-          subTree.push(grandChild);
-        });
+      if (child.children) {
+        if (expandedChildren.indexOf(child.data.id) !== -1 || extended) {
+          child.children.forEach(grandChild => {
+            subTree.push(grandChild);
+            if (expandedChildren.indexOf(child.data.id) !== -1 && extended && grandChild.children) {
+              child.children.forEach(extendedChild => {
```

Comment:
Doesn't this just att the granchild twice?

Reply:
Yes this should be grandChild.children.forEach I think

Reply:
yea it should, how the heck did this work then, or seem to work.

Reply:
I did not look closely at that =)